action.php

```diff
@@ -144,9 +145,39 @@ public function handleHtmlEditformOutput($event, $param) {
         $origtext = io_readWikiPage($file,$id);
 
         // Insert original page on the side
-        $form = & $event->data;
-        $pos = $form->findElementByType('wikitext');
-        if ($pos===false) return;
+        $form = $event->data;
+        if (is_a($form, \dokuwiki\Form\Form::class)) {
+            $pos = $form->findPositionByType('textarea'); // new
+            if ($pos===false) return;
+            $this->handleHtmlEditformOutputNG($form, $pos, $origtext);
+        } else {
+            $pos = $form->findElementByType('wikitext'); // old
+            if ($pos===false) return;
+            $this->handleHtmlEditformOutputLegacy($form, $pos, $origtext);
+        }
+    }
+
+    protected function handleHtmlEditformOutputNG($form, $pos, $origtext) {
+        // Before the wikitext...
+        $form->addElement(new dokuwiki\Form\TagOpenElement('div', array('id'=>'wrapper__wikitext','class'=>'hor')), $pos++);
+        // After the wikitext...
+        $pos++;
+        $form->addElement(new dokuwiki\Form\TagCloseElement('div'), $pos++);
+        $form->addElement(new dokuwiki\Form\TagOpenElement('div', array('id'=>'wrapper__sourcetext','class'=>'hor')), $pos++);
+        $origelem = '<textarea id="translate__sourcetext" '.
+                    //buildAttributes($attrs,true).
+                    'class="edit" readonly="readonly" cols="80" rows="10"'.
+                    'style="width:100%;"'.//  height:600px; overflow:auto
+                    '>'.NL.
+                    hsc($origtext).
+                    '</textarea>';
+        $form->addElement(new dokuwiki\Form\HTMLElement($origelem), $pos++);
+        $form->addElement(new dokuwiki\Form\TagCloseElement('div'), $pos++);
+        $form->addElement(new dokuwiki\Form\TagOpenElement('div', array('class'=>'clearer')), $pos++);
+        $form->addElement(new dokuwiki\Form\TagCloseElement('div'), $pos++);
+    }
+
+    protected function handleHtmlEditformOutputLegacy($form, $pos, $origtext) {
```

Comment:
Perhaps add a comment that this is used for dokuwiki < x.y.z (version)

Reply:
noted. will be fixed this evening

Reply:
done again